proposals/5/README.md

```diff
@@ -0,0 +1,103 @@
+****---
+title: Decomposition of Jenkins X
+linktitle: Decomposition of Jenkins X
+description: Decomposition of Jenkins X
+type: docs
+weight: 50
+---
+
+## 1. Overview
+
+This document outlines the work for decomposing Jenkins X into a series of reusable libraries.
+
+## 1.1 Motivation
+
+At the moment, the source code that comprises the main JX binary is tightly coupled and difficult to work on.  Changes have
+unintended consequences and this makes the codebase fragile.
+
+One of the design goals in Accelerate is to have a **Loosely Coupled Architecture** - in particular: 
+
+_The goal is for your architecture to support the ability of teams to get their work done—from design through to deployment—without requiring high-bandwidth communication between teams._
+
+## 1.2 Background
+
+The main [JX repository](https://github.com/jenkins-x/jx) contains over 250k lines of code, creating Apps / Extensions / Binary Plugins is difficult without adding a dependency back onto the JX repo.
+
+It should be possible to separate the repository out into a series of reusable libraries so only what is needed can be imported.
+
+## 2. Design
+
+There are currently a number of proposed repositories/modules, some of these are currently in flight.
+
+### Existing Modules
+
+* [jx-api](https://github.com/jenkins-x/jx-api): JX CRDs / Installation Requirements
```

Comment:
I realize - at least during the long decomposition transition phase - putting controllers into the api package doesn't work since we'd have circular dependency as long as we also move all required logic into packages into the `jx-api` repo as well.

Reply:
I am not sure whether this API library should initially contain InstallRequirements. It's really a different thing and since it is "just" a struct right now, I think other rules in terms of versioning and evolution apply opposed to a library of CRDs with their CRUD interfaces. 

Maybe long term it is the way to go, but I would not start of this was.